sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala

```diff
@@ -272,6 +244,44 @@ class HiveContext(sc: SparkContext) extends SQLContext(sc) {
         Nil
     }
 
+  override protected[sql] def createSession(): SQLSession = {
+    new this.SQLSession()
+  }
+
+  protected[hive] class SQLSession extends super.SQLSession {
```

Comment:
@guowei2 I think either way is OK for now. Putting all session-specific stuff into a central place (`SQLSession`) seems cleaner to me. Making `SQLSession` a thread-local does look a little ugly, however, right now it's not used anywhere other than the Thrift server. When we do decide to move Hive into a separate data source and make our own data source neutral Spark SQL server, we can handle the session problem in a cleaner way (e.g., using an actor for each session and keep all session-specific stuff in the actor instance).